src/strands/agui/bridge.py

```diff
@@ -0,0 +1,295 @@
+                            },
+                        )
+                    )
+
+                # Handle tool arguments
+                if "input" in tool_use:
+                    tool_input = tool_use["input"]
+                    if isinstance(tool_input, dict):
+                        tool_input = json.dumps(tool_input)
+                    elif not isinstance(tool_input, str):
+                        tool_input = str(tool_input)
+
+                    agui_events.append(
+                        self._create_agui_event(
+                            event_type=AGUIEventType.TOOL_CALL_ARGS, data={"tool_call_id": tool_id, "delta": tool_input}
+                        )
+                    )
+
+        # Handle completion events
+        if strands_event.get("complete", False):
+            # End any active tool calls
+            for tool_id in list(self.active_tool_calls.keys()):
+                agui_events.append(
+                    self._create_agui_event(event_type=AGUIEventType.TOOL_CALL_END, data={"tool_call_id": tool_id})
+                )
+                del self.active_tool_calls[tool_id]
+
+        return agui_events
+
+    def _on_state_change(self, new_state: Dict[str, Any], updates: Dict[str, Any]) -> None:
+        """Handle state changes from the state manager."""
+        # Queue state delta event
+        if updates:
+            state_event = self._create_agui_event(
+                event_type=AGUIEventType.STATE_DELTA, data={"delta": self._dict_to_json_patch(updates)}
+            )
+            self._state_change_queue.append(state_event)
+
+    def _dict_to_json_patch(self, updates: Dict[str, Any]) -> List[Dict[str, Any]]:
+        """Convert dictionary updates to JSON Patch format."""
+        patches = []
+        for key, value in updates.items():
+            if value is None:
+                patches.append({"op": "remove", "path": f"/{key}"})
+            else:
+                patches.append({"op": "replace", "path": f"/{key}", "value": value})
+        return patches
+
+    def _create_agui_event(self, event_type: AGUIEventType, data: Dict[str, Any]) -> Dict[str, Any]:
+        """Create a properly formatted AG-UI protocol event."""
+        return {"type": event_type.value, "timestamp": int(datetime.now().timestamp() * 1000), **data}
+
+
+class StrandsAGUIEndpoint:
+    """HTTP endpoint that serves AG-UI events from Strands agents with state management."""
+
+    def __init__(self, agents: Dict[str, Any]) -> None:
+        """Initialize the Strands AG-UI endpoint.
+
+        Args:
+            agents: Dictionary of agent name to agent instance
+        """
+        self.agents = agents
+        self.bridges: Dict[str, StrandsAGUIBridge] = {}
+
+        # Create bridges for each agent
+        for name, agent in agents.items():
+            self.bridges[name] = StrandsAGUIBridge(agent)
+
+    async def handle_request(self, request_data: Dict[str, Any]) -> AsyncIterator[str]:
```

Comment:
Remark: This might have changed a bit since this PR was opened. See https://docs.ag-ui.com/sdk/python/core/types#runagentinput. Notably, there is now a "tools" parameters defined in the AG-UI protocol used for frontend-tools.